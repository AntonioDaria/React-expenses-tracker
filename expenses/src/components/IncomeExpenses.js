import React, { useContext } from 'react'
import { GlobalContext } from '../context/GlobalState'
import {calculateIncome, calculateExpense} from "../utils/utils"

export const IncomeExpenses = () => {
    const { transactions } = useContext(GlobalContext)

    const amounts = transactions.map(transaction => { return transaction.amount })

    return (
        <div className="inc-exp-container">
            <div>
                <h4>Income</h4>
                <p className="money plus">+{calculateIncome(amounts)}</p>
            </div>
            <div>
                <h4>Expense</h4>
                <p className="money minus">-${calculateExpense(amounts)}</p>
            </div>
        </div>
    )
}



export default IncomeExpenses
