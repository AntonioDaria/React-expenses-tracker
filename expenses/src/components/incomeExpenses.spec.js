import React from 'react';
import { render } from '@testing-library/react';
import {IncomeExpenses}  from './IncomeExpenses';



describe('IncomeExpenses component', () => {
    let container;

    beforeEach(() => {
      ({ container} = renderIncomeExpenses());
    });

    it('should match the snapshot', () => {
        expect(container).toMatchSnapshot();
    });

    it.todo('should display the right amount for income');
    it.todo('should display the right amount for expense');
});

const renderIncomeExpenses= () => {
    return render(
      <IncomeExpenses/>
    );
  };
  