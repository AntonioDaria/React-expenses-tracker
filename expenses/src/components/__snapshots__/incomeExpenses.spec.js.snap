// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncomeExpenses component should match the snapshot 1`] = `
<div>
  <div
    class="inc-exp-container"
  >
    <div>
      <h4>
        Income
      </h4>
      <p
        class="money plus"
      >
        +
        0.00
      </p>
    </div>
    <div>
      <h4>
        Expense
      </h4>
      <p
        class="money minus"
      >
        -$
        0.00
      </p>
    </div>
  </div>
</div>
`;
